### ISSUE TITLE
Different success conditions in training and testing

### ISSUE THREAD

**Issue by williomma** [pengzhi1998/Robotics_Navigation#2]
Hi, thanks for your great work!

I noticed in unity_underwater_env.py that the success conditions for training and testing are different:

# Training
if self.training:
    if obs_goal_depthfromwater[0] < 0.6:
        reward_goal_reached = 10 - 8 * np.abs(obs_goal_depthfromwater[1]) - np.abs(np.deg2rad(obs_goal_depthfromwater[2]))
        done = True
        print("Reached the goal area!")
    else:
        reward_goal_reached = 0

# Testing
else:
    if obs_goal_depthfromwater[0] < 0.8:
        reward_goal_reached = 10 - 8 * np.abs(obs_goal_depthfromwater[1]) - np.abs(np.deg2rad(obs_goal_depthfromwater[2]))
        done = True
        print("Reached the goal area!")
    else:
        reward_goal_reached = 0

During training, the success threshold is obs_goal_depthfromwater[0] < 0.6, while during testing it is obs_goal_depthfromwater[0] < 0.8.

Could you please clarify why the scope of success is different in training and testing?

Or should they be consistent?

Thanks!

**Comment by pengzhi1998** [comment]
Hi, thank you for your interest in our work, and sorry for my delayed reply.

That’s a very good question! During training, we intentionally set a smaller success threshold because we want the agent to learn a more precise and efficient navigation policy. In contrast, during testing — especially in real-world experiments where the positioning system has lower accuracy — we use a slightly larger success range. This makes the evaluation more realistic and robust to sensor noise and uncertainty.
